python/ambassador/ir/irlistener.py

```diff
@@ -21,20 +21,25 @@ class IRListener (IRResource):
     IRListener is a pretty direct translation of the Ambassador Listener resource.
     """
 
+    socket_protocol: str # tcp or udp
```

Comment:
For enum variables, instead of typing it as `: str`, they can be typed as `: Literal["tcp", "udp"]` (given `from typing import Literal`).